### PR TITLE
Adds alias by URI and manual user inputs

### PR DESCRIPTION
**Pull request by khanek**

### CODE REVIEW

File: onetimepass/otp.py [hunk 1]
```diff
@@ -225,47 +252,100 @@ def import_(ctx: click.Context, file, format_: str):
...
+    uri_parsed = otp_auth_uri.parse(uri)
+    return create_alias_schema(
+        otp_type=uri_parsed.otp_type,
+        label=uri_parsed.label,
+        issuer=uri_parsed.issuer,
+        secret=uri_parsed.secret,
+        digits_count=uri_parsed.digits,
+        hash_algorithm=uri_parsed.algorithm,
+        params=uri_parsed.params,
+    )
```
Comment (Toreno96) [inline onetimepass/otp.py]: Maybe let's implement this as `AliasSchema.from_uri` classmethod?

File: onetimepass/otp.py [hunk 2]
```diff
@@ -225,47 +252,100 @@ def import_(ctx: click.Context, file, format_: str):
...
         )
         input_uri = click.prompt("Enter URI", confirmation_prompt=True, hide_input=True)
+        alias_data = get_alias_from_uri(input_uri)
     else:
         input_secret = click.prompt(
             "Enter secret", confirmation_prompt=True, hide_input=True
         )
+        params = (
+            {
+                "initial_time": datetime.datetime.fromtimestamp(
```
Comment (Toreno96) [inline onetimepass/otp.py]: Maybe let's extract this as the option w/ default? I think the user should be able to control this parameter, even if it's not supported by the Key URI schema.

File: onetimepass/otp.py [hunk 3]
```diff
@@ -283,10 +288,29 @@ def import_(ctx: click.Context, file, format_: str):
...
+@otp.group(
+    context_settings={"help_option_names": ["-h", "--help"]},
+    help="Add the new secret as the specified ALIAS.",
+)
+@click.pass_context
+def add(ctx: click.Context):
+    pass
+
+
+@add.command("uri", help="Add the new secret as the specified ALIAS.")
```
Comment (Toreno96) [inline onetimepass/otp.py]: I would suggest `help="Add the new secret from URI as the specified ALIAS."`, WDYT?

File: onetimepass/otp.py [hunk 4]
```diff
@@ -296,11 +320,100 @@ def get_alias_from_uri(uri: str) -> AliasSchema:
...
+    db = JSONEncryptedDB(path=settings.DB_PATH, key=keyring_get().encode())
+    data = get_db_data(db)
+    data.add_alias(alias, alias_data)
+    db.write(data)
+    if not quiet:
+        click.echo(f"{alias} added")
+
 
-@otp.command(help="Add the new secret as the specified ALIAS.")
+@add.command("hotp")
```
Comment (Toreno96) [inline onetimepass/otp.py]: I would suggest adding `help="Add the new HOTP secret as the specified ALIAS."`, WDYT?

File: onetimepass/otp.py [hunk 5]
```diff
@@ -310,69 +423,42 @@ def get_alias_from_uri(uri: str) -> AliasSchema:
...
-    type=click.Choice(OTPType),
-    show_default=True,
-    default=OTPType.TOTP.value,
+    "--initial-time",
+    type=click.DateTime(),
+    default=get_default_initial_time,
 )
 @click.pass_context
-def add(
+def add_totp(
```
Comment (Toreno96) [inline onetimepass/otp.py:420]: I would suggest adding `help="Add the new TOTP secret as the specified ALIAS."`, WDYT?

File: onetimepass/otp.py [hunk 6]
```diff
@@ -283,10 +288,29 @@ def import_(ctx: click.Context, file, format_: str):
...
+@add.command("uri", help="Add the new secret as the specified ALIAS.")
+@click.argument("alias")
+@click.pass_context
+def add_uri(ctx: click.Context, alias: str):
+    quiet = ctx.obj["quiet"]
+
+    db = JSONEncryptedDB(path=settings.DB_PATH, key=keyring_get().encode())
+    data = get_db_data(db)
+    if alias in data.otp:
+        raise click.UsageError(f"Alias {alias} exists. Consider renaming it")
```
Comment (Toreno96) [inline onetimepass/otp.py]: Please use the new `ClickUsageError` wrapper instead.

File: onetimepass/otp.py [hunk 7]
```diff
@@ -296,11 +320,100 @@ def get_alias_from_uri(uri: str) -> AliasSchema:
...
+    "algorithm",
+    "-a",
+    "--algorithm",
+    show_default=True,
+    type=click.Choice(OTPAlgorithm),
+    default=OTPAlgorithm.SHA1.value,
+)
+@click.option(
+    "digits_count",
+    "-dc",
```
Comment (Toreno96) [inline onetimepass/otp.py]: I think we could just use `-d`

File: onetimepass/otp.py (hunk 5 above)
Comment (khanek) [reply onetimepass/otp.py:420]: done

**Review by Toreno96 — APPROVED** [review]

**Pull request merged**